test/NuGet.Services.Staging.Test.EndToEnd/StagingClient/StagingClient.cs

```diff
@@ -164,17 +164,36 @@ public StagingClient(Uri stagingServiceUri, ITestOutputHelper logger)
             return JObject.Parse(responseBody);
         }
 
-        public async Task<JObject> Query(string stageId, string queryString)
+        public async Task<JObject> Query(string stageId, string query, bool includePrerelease=false, int skip=0, int take=20)
         {
             _logger.WriteLine($"StagingClient: Query called for stage {stageId}");
 
             JObject index = await Index(stageId);
 
-            string searchEndpoint = index["resources"].Where(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0]).ToString();
+            string searchEndpoint = index["resources"].First(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0])["@id"].ToString();

```

Comment:
Should we handle the potential nullref that would happen on this line if the service type does not exist in the index? Or has the service index been validated elsewhere already, so we can assume all required service types are present throughout the code?